modules/serverless-tag-to-digest-resolution.adoc

```diff
@@ -0,0 +1,52 @@
+// Module included in the following assemblies
+//
+// * serverless/admin_guide/serverless-configuration.adoc
+
+:_content-type: CONCEPT
+[id="serverless-tag-to-digest-resolution_{context}"]
+= Configuring tag-to-digest resolution
+
+If the Knative Serving controller has access to the container registry, Knative Serving resolves image tags to a digest when you create a revision of a service. This is referred to as _tag-to-digest resolution_, and helps to provide consistency for deployments.
+
+If your container registry uses a self-signed certificate, you must configure the Knative Serving controller so that it trusts that certificate. To do this, you must create a config map or secret and then configure controller custom certificates by modifying the `controller-custom-certs` spec in the `KnativeServing` custom resource (CR). The config map or secret must reside in the same namespace as the Knative Serving CR. If no controller custom certificate is specified, this setting defaults to use the `config-service-ca` config map.
+
+[id="knative-serving-controller-custom-certs-secrets_{context}"]
+== Using a secret
+
+If the type configured in the `controller-custom-certs` spec is `secret`, the {ServerlessOperatorName} takes no action. The secret is mounted as a secret volume, and Knative components consume the secret directly, assuming that the secret contains the required certificates.
+// example YAML?
```

Comment:
I dont have a setup for this not sure if has been tested before, @mgencur or @maschmid ?

Reply:
AFAICT, this hasn't been tested on our side. I can't find any tests for this so I don't have a good example config.

Reply:
We created https://issues.redhat.com/browse/SRVCOM-1860 to test this. But I'm not sure how quickly we can do it.

Reply:
The example yaml would be this:
```
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeServing
metadata:
  name: knative-serving
  namespace: knative-serving
spec:
  controller-custom-certs:
    name: custom-secret
    type: Secret
```
This would also require the secret to be created in advance by running this:
```
oc -n knative-serving create secret generic custom-secret --from-file=foo.crt=<path/to/certificate> 
```
(The name "foo" is arbitrary)

We have a test for this now: [link](https://gitlab.cee.redhat.com/ocf-qe/ocf-qe-testsuite/-/blob/master/test/operator/digest_resolution_test.go#L117)

Reply:
Probably need to split secret and config map into two separate procedure modules in this case, since there are a few steps required to use this (it can't really be considered reference / conceptual docs then). This is mostly a note for myself 🙂